src/vmm/src/devices/virtio/vsock/csm/connection.rs

```diff
@@ -85,17 +85,23 @@ use std::time::{Duration, Instant};
 
 use logger::{debug, error, info, warn, IncMetric, METRICS};
 use utils::epoll::EventSet;
-use utils::vm_memory::{GuestMemoryError, GuestMemoryMmap};
+use utils::vm_memory::{GuestMemoryError, GuestMemoryMmap, ReadVolatile, WriteVolatile};
 
 use super::super::defs::uapi;
 use super::super::packet::VsockPacket;
 use super::super::{Result as VsockResult, VsockChannel, VsockEpollListener, VsockError};
 use super::txbuf::TxBuf;
 use super::{defs, ConnState, Error, PendingRx, PendingRxSet, Result};
 
+/// Trait that vsock connection backends need to implement.
+///
+/// Used as an alias for `Read + ReadVolatile + Write + WriteVolatile + AsRawFd` (sadly, trait
+/// aliases are not supported, https://github.com/rust-lang/rfcs/pull/1733#issuecomment-243840014).
+pub trait VsockConnectionBackend: Read + ReadVolatile + Write + WriteVolatile + AsRawFd {}
```

Comment:
Do we still need the `Read` and `Write` specializations here?

Reply:
Ahh, good catch! We do need `Write` due to the TxBuf -> Socket transfer happening in VMM-space (e.g. not through volatile memory), but the `Read` is indeed not needed anymore!